apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/context/ExtensionContext.java

```diff
@@ -49,11 +64,23 @@ void deserialize(String value) {
         if (StringUtil.isEmpty(value)) {
             return;
         }
-        final String[] extensionParts = value.split("-");
+        String[] extensionParts = value.replaceAll("-", "- ").split("-");
```

Comment:
Please inject ` `(whitespace), rather than this strange `replaceAll`.

Reply:
I may have misunderstood what you meant, thinking you meant to pass as few characters as possible. fix it later.

Reply:
The size is not a concern, as we have injected a much bigger `sw8` header.